include/materials/material_utility.h

```diff
@@ -0,0 +1,68 @@
+#ifndef MPM_MATERIAL_UTILITY_H_
+#define MPM_MATERIAL_UTILITY_H_
+
+#include <cmath>
+
+#include "Eigen/Dense"
+
+namespace mpm {
+namespace material_utility {
+
+//! Compute mean stress p
+//! \param[in] stress Stress in Voigt notation where positive is tension
+inline const double compute_mean_p(const Eigen::Matrix<double, 6, 1>& stress);
```

Comment:
Shall we rename the function as `p` and drop the `compute_` part from all the function names?

Reply:
I'm in two minds about the namespace `material_utility` instead of just `materials` because every time we use it, we have to do `mpm::material_utility::compute_` instead of `mpm::material::mean_p`, which would be more succinct to read.

Reply:
I understand. Should the filename be changed too? I will keep the filename as is for now

Reply:
Material class is a bit off I guess. I am still for material utility. Btw, @kks32 for the namespace are using CamelCase or snake_case?